crates/noirc_evaluator/src/ssa/context.rs

```diff
@@ -702,7 +702,7 @@ impl SsaContext {
         //Optimization
         block::compute_dom(self);
         optimizations::full_cse(self, self.first_block, false)?;
-
+        optimizations::full_cse(self, self.first_block, false)?;
```

Comment:
Can you elaborate on why we need to do a full_cse twice now?

Reply:
This one is not required, but investigating changing results with the Poseidon PR, I noticed that it can be help a lot. It is expected that a cse round generates opportunities for more optimisations, so ideally we would do it until there is no gain, but I am a bit worried on the performance impact and would prefer to leave this to an optimisation setting.